packages/components/pie-modal/src/index.ts

```diff
@@ -251,6 +258,41 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
         `;
     }
 
+    /**
+     * Render supportingAction button depending on prop availability.
+     *
+     * 1. If the prop `supportingAction` is not provided, the button is not rendered.
+     * 2. If the prop `supportingAction` is provided but any of the optional properties
+     * are not provided, they fall back to their default values.
+     * 3. If `supportingAction` is provided but not `leadingAction`, log a warning and do
+     * not render `supportingAction`.
+     *
+     * @private
+     */
+    private renderSupportingAction (): TemplateResult | typeof nothing {
+        const { text, variant = 'ghost', ariaLabel } = this.supportingAction;
+
+        if (!text) {
+            return nothing;
+        }
+
+        if (!this.leadingAction) {
+            console.warn('Use `leadingAction` instead of `supportingAction`. `supportingAction` is being ignored.');
+            return nothing;
+        }
+
+        return html`
+              <pie-button
+                variant="${variant}"
+                aria-label="${ariaLabel || nothing}"
```

Comment:
can we use ifDefined here?
https://lit.dev/docs/templates/directives/#ifdefined

Reply:
I tried a few things before i went this route initially. The issue is with `ifDefined` and `guard` they still render the attribute itself if the value is empty i.e `ariaLabel: ''`. With `nothing` if the ariaLabel is empty or doesn't exist then the attribute won't render.